### PR TITLE
gh-94512: Fix forced arg format in AC-processed resource

### PR DESCRIPTION
**Pull request by arhadthedev**
Please note that this commit also removes an old way to express optional parameters (it prevented the new style generation):

```diff
 /*[clinic input]
 resource.prlimit

     pid: pid_t
     resource: int
-    [
-    limits: object
-    ]
+    limits: object = None
     /

[clinic start generated code]*/
```

I hope it breaks nothing.

* Issue: gh-94512

### CODE REVIEW

**Comment by arhadthedev** [comment]
I have no idea why `python -m unittest -v test.test_resource.ResourceTest.test_prlimit` fails to import `resource` module (I didn't touch a startup function):

```text
Running Release|x64 interpreter...
Traceback (most recent call last):
  File "C:\Users\oleg\Documents\dev\notmine\cpython\Lib\test\support\import_helper.py", line 78, in import_module
    return importlib.import_module(name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\oleg\Documents\dev\notmine\cpython\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1142, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'resource'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\oleg\Documents\dev\notmine\cpython\Lib\unittest\__main__.py", line 18, in <module>
    main(module=None)
    ^^^^^^^^^^^^^^^^^
  File "C:\Users\oleg\Documents\dev\notmine\cpython\Lib\unittest\main.py", line 101, in __init__
    self.parseArgs(argv)
    ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\oleg\Documents\dev\notmine\cpython\Lib\unittest\main.py", line 150, in parseArgs
    self.createTests()
    ^^^^^^^^^^^^^^^^^^
  File "C:\Users\oleg\Documents\dev\notmine\cpython\Lib\unittest\main.py", line 161, in createTests
    self.test = self.testLoader.loadTestsFromNames(self.testNames,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\oleg\Documents\dev\notmine\cpython\Lib\unittest\loader.py", line 198, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\oleg\Documents\dev\notmine\cpython\Lib\unittest\loader.py", line 198, in <listcomp>
    suites = [self.loadTestsFromName(name, module) for name in names]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\oleg\Documents\dev\notmine\cpython\Lib\unittest\loader.py", line 132, in loadTestsFromName
    module = __import__(module_name)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\oleg\Documents\dev\notmine\cpython\Lib\test\test_resource.py", line 9, in <module>
    resource = import_helper.import_module('resource')
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\oleg\Documents\dev\notmine\cpython\Lib\test\support\import_helper.py", line 82, in import_module
    raise unittest.SkipTest(str(msg))
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
unittest.case.SkipTest: No module named 'resource'
```

**Comment by arhadthedev** [comment]
The bug is fixed.

File: Misc/NEWS.d/next/Library/2022-07-03-23-09-45.gh-issue-94512.9wtOYT.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+:meth:`resource.prlimit` changed its signature from
+``prlimit(pid, resource, [limits], /)`` to
+``prlimit($module, pid, resource, limits=None, /)``.
```
Comment (zooba) [inline Misc/NEWS.d/next/Library/2022-07-03-23-09-45.gh-issue-94512.9wtOYT.rst]: Doesn't require a NEWS entry for this, as the signature hasn't _really_ changed (`None` wasn't valid for limits before and its behaviour now is sensible)

We also shouldn't put implicit arguments in user-facing documentation.

Comment (arhadthedev) [reply Misc/NEWS.d/next/Library/2022-07-03-23-09-45.gh-issue-94512.9wtOYT.rst]: Addressed.

**Pull request merged**